src/types/message/index.js

```diff
@@ -154,11 +194,18 @@ class BitswapMessage {
     return Message.encode(msg)
   }
 
+  /**
+   * @param {BitswapMessage} other
+   * @returns {boolean}
+   */
   equals (other) {
     if (this.full !== other.full ||
         this.pendingBytes !== other.pendingBytes ||
         !isMapEqual(this.wantlist, other.wantlist) ||
         !isMapEqual(this.blocks, other.blocks) ||
+        // @TODO - Is this a bug ? `isMapEqual` only compare maps of blocks and
+        // wants
+        // @ts-ignore
```

Comment:
This one seems like a bug, but maybe it is not 🙄 So `isMapEqual` will compare only that keys match and will compare values if they are either `Block`s or `BitswapMessageEntry`s (anything that has `equals` method on them actually). However `blockPresences`  contain numbers so two would be equal even if the actual values are different, which doesn't seems  like intended behavior.

P.S.: TS not happy mostly because it's told to expect maps of blocks or entries and is passed maps of numbers below.